packages/legend-query/src/components/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -0,0 +1,641 @@
+    ): void => {
+      if (explorerState.treeData) {
+        setAnchorEl(event.currentTarget);
+        propertySearchPanelState.setIsSearchPanelOpen(true);
+        if (!propertySearchPanelState.allMappedPropertyNodes.length) {
+          propertySearchPanelState.fetchAllPropertyNodes();
+        }
+      }
+    };
+
+    return (
+      <>
+        <button
+          className="panel__header__action"
+          onClick={searchProperty}
+          tabIndex={-1}
+          title="Search for property"
+        >
+          <SearchIcon />
+        </button>
+        <BasePopover
+          open={propertySearchPanelState.isSearchPanelOpen}
+          anchorEl={anchorEl}
+          onClose={handleClose}
+          anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'left',
+          }}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'center',
+          }}
+        >
+          <div
+            data-testid={
+              QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PROPERTY_SEARCH_PANEL
+            }
+            className="query-builder-property-search-panel"
+          >
+            <div className="query-builder-property-search-panel__header">
+              <input
+                className={clsx(
+                  'query-builder-property-search-panel__input input--dark',
+                  {
+                    'query-builder-property-search-panel__input--searching':
+                      propertySearchPanelState.searchedPropertyName,
+                  },
+                )}
+                onChange={changePropertyName}
+                value={propertySearchPanelState.searchedPropertyName}
+                placeholder="Search for a property"
+              />
+              {!propertySearchPanelState.searchedPropertyName ? (
+                <div className="query-builder-property-search-panel__input__search__icon">
+                  <SearchIcon />
+                </div>
+              ) : (
+                <>
+                  <div className="query-builder-property-search-panel__input__search__count">
+                    {propertySearchPanelState.filteredPropertyNodes.length}
+                  </div>
+                  <button
+                    className="query-builder-property-search-panel__input__clear-btn"
+                    tabIndex={-1}
+                    onClick={clearPropertyName}
+                    title="Clear"
+                  >
+                    <TimesIcon />
+                  </button>
+                </>
+              )}
+            </div>
+            <div className="query-builder-property-search-panel__content">
+              <ResizablePanelGroup orientation="vertical">
+                <ResizablePanel size={150}>
+                  <div className="query-builder-property-search-panel__form__section">
+                    <div className="query-builder-property-search-panel__form__section__header__label">
+                      One many rows
```

Comment:
--> `Multiple` because we change the text, let's add a hint icon to explain what this is about. I don't think `One-many rows` is descriptive enough